### ISSUE TITLE
Codeship install issue

### ISSUE THREAD

**Issue by peteringram0** [jspm/jspm-cli#1024]
Hi Guys,

Any idea why im getting this issue when running jspm install on Codeship?

```node
Looking up npm:jsonp
Looking up npm:lodash
Creating registry cache...
Downloading npm:jsonp@0.2.0
Downloading npm:lodash@3.10.1
Looking up npm:debug
Downloading npm:debug@2.1.3
Looking up npm:ms
Downloading npm:ms@0.7.0
ok   Installed npm:debug@2.1.3 (2.1.3)
ok   Installed npm:ms@0.7.0 (0.7.0)
Looking up github:jspm/nodelibs-tty
Looking up github:jspm/nodelibs-util
Looking up github:jspm/nodelibs-process
Looking up github:jspm/nodelibs-fs
Looking up github:jspm/nodelibs-net
Downloading github:jspm/nodelibs-tty@0.1.0
Downloading github:jspm/nodelibs-fs@0.1.2
Downloading github:jspm/nodelibs-process@0.1.1
Downloading github:jspm/nodelibs-util@0.1.0
Downloading github:jspm/nodelibs-net@0.1.2
Looking up npm:process
Looking up npm:tty-browserify
Looking up npm:util

warn Error on download for github:jspm/nodelibs-fs, retrying (1).
Error: socket hang up
at createHangUpError (http.js:1472:15)
at CleartextStream.socketCloseListener (http.js:1522:23)
at CleartextStream.emit (events.js:117:20)
at tls.js:693:10
at process._tickCallback (node.js:419:13)

warn Error on download for github:jspm/nodelibs-util, retrying (1).
Error: socket hang up
at createHangUpError (http.js:1472:15)
at CleartextStream.socketCloseListener (http.js:1522:23)
at CleartextStream.emit (events.js:117:20)
at tls.js:693:10
at process._tickCallback (node.js:419:13)

warn Error on download for github:jspm/nodelibs-tty, retrying (1).
Error: socket hang up
at createHangUpError (http.js:1472:15)
at CleartextStream.socketCloseListener (http.js:1522:23)
at CleartextStream.emit (events.js:117:20)
at tls.js:693:10
at process._tickCallback (node.js:419:13)

warn Error on download for github:jspm/nodelibs-net, retrying (1).
Error: socket hang up
at createHangUpError (http.js:1472:15)
at CleartextStream.socketCloseListener (http.js:1522:23)
at CleartextStream.emit (events.js:117:20)
at tls.js:693:10
at process._tickCallback (node.js:419:13)

warn Error on download for github:jspm/nodelibs-process, retrying (1).
Bad response code 502
Downloading npm:util@0.10.3
Looking up npm:inherits

warn Error on download for github:jspm/nodelibs-fs, retrying (2).
Error: socket hang up
at createHangUpError (http.js:1472:15)
at CleartextStream.socketCloseListener (http.js:1522:23)
at CleartextStream.emit (events.js:117:20)
at tls.js:693:10
at process._tickCallback (node.js:419:13)

warn Error on download for github:jspm/nodelibs-net, retrying (2).
Error: socket hang up
at createHangUpError (http.js:1472:15)
at CleartextStream.socketCloseListener (http.js:1522:23)
at CleartextStream.emit (events.js:117:20)
at tls.js:693:10
at process._tickCallback (node.js:419:13)

warn Error on download for github:jspm/nodelibs-util, retrying (2).
Bad response code 502

warn Error on download for github:jspm/nodelibs-process, retrying (2).
Error: socket hang up
at createHangUpError (http.js:1472:15)
at CleartextStream.socketCloseListener (http.js:1522:23)
at CleartextStream.emit (events.js:117:20)
at tls.js:693:10
at process._tickCallback (node.js:419:13)
Downloading npm:process@0.10.1
Downloading npm:tty-browserify@0.0.0

warn Error on download for github:jspm/nodelibs-util, retrying (3).
Bad response code 502

warn Error on download for github:jspm/nodelibs-process, retrying (3).
Error: socket hang up
at createHangUpError (http.js:1472:15)
at CleartextStream.socketCloseListener (http.js:1522:23)
at CleartextStream.emit (events.js:117:20)
at tls.js:693:10
at process._tickCallback (node.js:419:13)
Downloading npm:inherits@2.0.1
ok   Installed npm:tty-browserify@0.0.0 (0.0.0)
ok   Installed npm:process@^0.10.0 (0.10.1)

warn Error on download for github:jspm/nodelibs-util
Error: socket hang up
at createHangUpError (http.js:1472:15)
at CleartextStream.socketCloseListener (http.js:1522:23)
at CleartextStream.emit (events.js:117:20)
at tls.js:693:10
at process._tickCallback (node.js:419:13)

err  Error downloading github:jspm/nodelibs-util.

warn Installation changes not saved.
```

**Comment by guybedford** [comment]
Are you perhaps behind a proxy here? It seems that your `github` installs aren't working. Are you able to install for example `jspm install jquery`? Perhaps look at setting the proxy environment variables here?

**Comment by guybedford** [comment]
(I was definitely able to install this fine)

**Comment by peteringram0** [comment]
This is on Codeship so codeship is executing these commands im not sure if its behind a proxy or not but i dont expect so.

**Comment by guybedford** [comment]
Is there anyway you can share a replication scenario here at all then?

**Comment by guybedford** [comment]
Let me know if there is anything I can do to help here.

**Comment by Rush** [comment]
I also have problems with codeship, even after setting an API key it's failing regularly, preventing our CI to do its job ... any ideas aside of commiting jspm_packages to our repo?

```

warn GitHub rate limit reached.

Would you like to set up your GitHub credentials? [yes]:ok   Installed npm:string_decoder@~0.10.0 (0.10.31)
ok   Installed npm:isarray@0.0.1 (0.0.1)
ok   Installed npm:Base64@~0.2.0 (0.2.1)
Downloading github:jspm/nodelibs-buffer@0.1.0
ok   Installed github:jspm/nodelibs-assert@^0.1.0 (0.1.0)
ok   Installed npm:core-util-is@~1.0.0 (1.0.1)
ok   Installed github:jspm/nodelibs-http@^1.7.0 (1.7.1)
ok   Installed npm:optimist@~0.3.5 (0.3.7)
ok   Installed npm:assert@^1.3.0 (1.3.0)
ok   Installed npm:source-map@~0.4.0 (0.4.4)
ok   Installed npm:events@1.0.2 (1.0.2)
ok   Installed npm:readable-stream@^1.0.27-1 (1.1.13)
ok   Installed npm:util@0.10.3 (0.10.3)
ok   Installed npm:source-map@0.1.34 (0.1.34)
ok   Installed npm:uglify-js@~2.2.5 (2.2.5)
ok   Installed npm:clean-css@^3.1.9 (3.4.1)
ok   Installed npm:uglify-js@^2.4.19 (2.4.24)
ok   Installed npm:jade@1.11.0 (1.11.0)
ok   Installed npm:acorn@^2.1.0 (2.3.0)
ok   Installed babel-runtime as npm:babel-runtime@^5.8.20 (5.8.20)
ok   Installed npm:acorn@^1.0.1 (1.2.2)
ok   Installed babel as npm:babel-core@^5.8.22 (5.8.23)

warn Timed out on getPackageConfig for github:jspm/nodelibs-util, retrying (1).
To increase the timeout run jspm config registries.github.timeouts.lookup 120

warn Timed out on getPackageConfig for github:jspm/nodelibs-buffer, retrying (1).
To increase the timeout run jspm config registries.github.timeouts.lookup 120

warn Timed out on locate for github:jspm/nodelibs-vm, retrying (1).
To increase the timeout run jspm config registries.github.timeouts.lookup 120

warn Timed out on locate for github:jspm/nodelibs-url, retrying (1).
To increase the timeout run jspm config registries.github.timeouts.lookup 120

warn Timed out on locate for github:jspm/nodelibs-https, retrying (1).
To increase the timeout run jspm config registries.github.timeouts.lookup 120

warn Timed out on locate for github:jspm/nodelibs-os, retrying (1).
To increase the timeout run jspm config registries.github.timeouts.lookup 120

warn Timed out on locate for github:systemjs/plugin-json, retrying (1).
To increase the timeout run jspm config registries.github.timeouts.lookup 120

warn Timed out on getPackageConfig for github:jspm/nodelibs-util, retrying (2).
To increase the timeout run jspm config registries.github.timeouts.lookup 120

warn Timed out on getPackageConfig for github:jspm/nodelibs-buffer, retrying (2).
To increase the timeout run jspm config registries.github.timeouts.lookup 120

warn Timed out on locate for github:jspm/nodelibs-vm, retrying (2).
To increase the timeout run jspm config registries.github.timeouts.lookup 120

warn Timed out on locate for github:jspm/nodelibs-url, retrying (2).
To increase the timeout run jspm config registries.github.timeouts.lookup 120

warn Timed out on locate for github:jspm/nodelibs-https, retrying (2).
To increase the timeout run jspm config registries.github.timeouts.lookup 120

warn Timed out on locate for github:jspm/nodelibs-os, retrying (2).
To increase the timeout run jspm config registries.github.timeouts.lookup 120

warn Timed out on locate for github:systemjs/plugin-json, retrying (2).
To increase the timeout run jspm config registries.github.timeouts.lookup 120

warn Timed out on getPackageConfig for github:jspm/nodelibs-util, retrying (3).
To increase the timeout run jspm config registries.github.timeouts.lookup 120

warn Timed out on getPackageConfig for github:jspm/nodelibs-buffer, retrying (3).
To increase the timeout run jspm config registries.github.timeouts.lookup 120

warn Timed out on locate for github:jspm/nodelibs-vm, retrying (3).
To increase the timeout run jspm config registries.github.timeouts.lookup 120

warn Timed out on locate for github:jspm/nodelibs-url, retrying (3).
To increase the timeout run jspm config registries.github.timeouts.lookup 120

warn Timed out on locate for github:jspm/nodelibs-https, retrying (3).
To increase the timeout run jspm config registries.github.timeouts.lookup 120

warn Timed out on locate for github:jspm/nodelibs-os, retrying (3).
To increase the timeout run jspm config registries.github.timeouts.lookup 120

warn Timed out on locate for github:systemjs/plugin-json, retrying (3).
To increase the timeout run jspm config registries.github.timeouts.lookup 120

warn Timed out on getPackageConfig for github:jspm/nodelibs-util
To increase the timeout run jspm config registries.github.timeouts.lookup 120

warn Timed out on getPackageConfig for github:jspm/nodelibs-buffer
To increase the timeout run jspm config registries.github.timeouts.lookup 120

warn Error getting package config for github:jspm/nodelibs-util.

warn Error getting package config for github:jspm/nodelibs-buffer.

warn Timed out on locate for github:jspm/nodelibs-vm
To increase the timeout run jspm config registries.github.timeouts.lookup 120

err  Error locating github:jspm/nodelibs-vm.

warn Installation changes not saved.
```

**Comment by kevinmpowell** [comment]
@Rush here's how I have it configured:

`JSPM_GITHUB_AUTH_TOKEN` is configured as an environment variable on Codeship:
![screen shot 2015-09-14 at 1 56 53 pm](https://cloud.githubusercontent.com/assets/979463/9858858/a9444e76-5ae8-11e5-8b26-7f4ad476746b.png)

Then referenced in the Setup Commands on Codeship:
![screen shot 2015-09-14 at 1 59 44 pm](https://cloud.githubusercontent.com/assets/979463/9858892/e2be8f40-5ae8-11e5-83fa-25b82865bebe.png)


I set this up several months ago when I was getting those same Github rate limit errors and I haven't had any problems since.

**Comment by Rush** [comment]
Thanks a bunch, is there a difference between `endpoints.github.auth` and `registries.github.auth`?

**Comment by guybedford** [comment]
@Rush `registries` used to be called `endpoints`, but that term was deprecated to reduce confusion.

**Comment by gillyd12** [comment]
@kevinmpowell Did you change the github scopes with that setup?  I am running into an Unauthorized error from Github during my CodeShip setup step jspm install and followed these instructions.  Its pretty straight forward, surprised its causing me this much trouble.

**Comment by mpint** [comment]
@gillyd12, I got this working.

Use this as a guide.
http://www.sanjeevnandam.com/blog/how-to-fix-jspm-github-limit-reached-error-in-a-ci-server

First, log into your codeship server by "Open SSH Debug Session" which is a link on the page: https://codeship.com/projects/<YOUR_PROJECT>/builds/<SOME_BUILD>

When you generate your github token, you must check `repo`, or at least `public_repo` (I didn't test whether checking only `public_repo` worked).

Then, proceed with what @kevinmpowell outlined in his post, making sure `JSPM_GITHUB_AUTH_TOKEN` is not actually the github personal token you generated, but the result of the sequence of commands in the video I linked above.

Hope this explanation saves a few headaches.
